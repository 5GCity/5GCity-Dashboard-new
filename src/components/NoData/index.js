/**
 * NoData Component
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { NoDataIcon } from 'components/Icons'

export default ({ children, ...props }) => (
  <Wrapper {...props}>
    <SVG><NoDataIcon /></SVG>
    <Title>{props.title}</Title>
    <Message>{props.message}</Message>
  </Wrapper>
)

const Wrapper = styled.div`
top: 150px;
    left: 0;
    right: 0;
    margin-left: auto;
    margin-right: auto;
    position: relative;
`
const SVG = styled.div`
  text-align: center;
`
const Title = styled.h1`
  color: #EFF2F7;
  font-family: ${({theme}) => theme.fontDin};
  font-size: 24px;
  line-height: 32px;
  letter-spacing: 0.6px;
  text-align: center;
`
const Message = styled.h2`
  color: ${({theme}) => theme.secondaryColor};
  font-family: ${({theme}) => theme.fontFamily};
  font-size: 16px;
  line-height: 16px;
  font-weight: 600;
  text-align: center;
`
