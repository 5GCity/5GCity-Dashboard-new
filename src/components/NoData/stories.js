/**
 * NoData Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'

import NoData from './index'

const State1 = () => (
  <NoData
    message={'Click on the “Add new service” button to add your first service!'}
    title={'You don’t have any services yet...'}
  />
)

storiesOf('NoData', module)
  .add('primary', State1)
