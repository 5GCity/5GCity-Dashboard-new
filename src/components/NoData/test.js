/**
 * NoData Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import NoData from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<NoData />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
