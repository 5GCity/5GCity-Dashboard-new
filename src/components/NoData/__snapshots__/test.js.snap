// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: block;
  margin-top: 150px;
}

.c1 {
  text-align: center;
}

.c2 {
  color: #EFF2F7;
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  text-align: center;
}

.c3 {
  font-size: 16px;
  line-height: 16px;
  font-weight: 600;
  text-align: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      height={240}
      width={252}
    >
      <defs>
        <linearGradient
          id="prefix__a"
          x1="49.999%"
          x2="49.999%"
          y1="100%"
          y2="0%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
        <linearGradient
          id="prefix__b"
          x1="50%"
          x2="50%"
          y1="100%"
          y2="0%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
        <linearGradient
          id="prefix__c"
          x1="60.274%"
          x2="39.783%"
          y1="98.88%"
          y2="1.029%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
        <linearGradient
          id="prefix__d"
          x1="49.999%"
          x2="49.999%"
          y1="100%"
          y2="0%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
        <linearGradient
          id="prefix__e"
          x1="50%"
          x2="50%"
          y1="100%"
          y2="0%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
        <linearGradient
          id="prefix__f"
          x1="50%"
          x2="50%"
          y1="99.914%"
          y2="-.086%"
        >
          <stop
            offset="0%"
            stopColor="gray"
            stopOpacity={0.25}
          />
          <stop
            offset="54%"
            stopColor="gray"
            stopOpacity={0.12}
          />
          <stop
            offset="100%"
            stopColor="gray"
            stopOpacity={0.1}
          />
        </linearGradient>
      </defs>
      <g
        fill="none"
        fillRule="nonzero"
      >
        <path
          d="M21.216 41.636L152.37 13.462l38.882 181.424-131.155 28.173z"
          fill="#89979F"
        />
        <path
          d="M17.51 11.74l129.613-.03-.043 179.271-129.613.03z"
          fill="url(#prefix__a)"
          opacity={0.5}
          transform="rotate(-12.11 173.933 -3.04)"
        />
        <path
          d="M25.98 45.17l123.495-26.528 36.738 171.419-123.496 26.528z"
          fill="#5A666D"
        />
        <path
          d="M56.56 24.28l62.662-.016-.004 15.196-62.662.014z"
          fill="url(#prefix__b)"
          transform="rotate(-12.11 87.889 31.87)"
        />
        <path
          d="M83.185 9.929c-4.73 1.017-7.723 5.77-6.683 10.617 1.04 4.846 5.716 7.956 10.436 6.939 4.721-1.017 7.723-5.771 6.683-10.617-1.04-4.846-5.706-7.956-10.436-6.94zm2.968 13.859a5.172 5.172 0 0 1-5.406-2.02 5.186 5.186 0 0 1-.177-5.775 5.173 5.173 0 0 1 9.438 1.649 5.125 5.125 0 0 1-.652 3.875 5.114 5.114 0 0 1-3.203 2.271z"
          fill="url(#prefix__c)"
        />
        <path
          d="M56.173 30.245l59.974-12.883 3.044 14.202-59.974 12.884z"
          fill="#8CC14E"
        />
        <path
          d="M83.185 9.929c-4.63.995-7.577 5.558-6.583 10.192.993 4.635 5.55 7.586 10.18 6.591 4.63-.994 7.577-5.557 6.584-10.192a8.583 8.583 0 0 0-3.727-5.407 8.565 8.565 0 0 0-6.454-1.184zm2.836 13.244a4.945 4.945 0 0 1-5.11-2.03 4.958 4.958 0 0 1-.083-5.503 4.946 4.946 0 0 1 8.993 1.652 4.954 4.954 0 0 1-3.8 5.881z"
          fill="#8CC14E"
        />
        <path
          d="M79.157 35.926h134.14v185.553H79.157z"
          fill="#89979F"
        />
        <path
          d="M.086.037h129.607v179.281H.086z"
          fill="url(#prefix__d)"
          opacity={0.5}
          transform="translate(81.34 39.027)"
        />
        <path
          d="M83.074 40.385h126.307v175.32H83.074z"
          fill="#5A666D"
        />
        <path
          d="M115.064 32.783h62.659v15.196h-62.659z"
          fill="url(#prefix__e)"
        />
        <path
          d="M146.393 17.95c-4.837 0-8.76 4.019-8.76 8.979 0 4.96 3.923 8.979 8.76 8.979 4.838 0 8.76-4.02 8.76-8.98 0-4.96-3.922-8.979-8.76-8.979zm0 14.16a5.175 5.175 0 0 1-4.867-3.112 5.186 5.186 0 0 1 1.042-5.687 5.17 5.17 0 0 1 5.653-1.18 5.181 5.181 0 0 1 3.225 4.798 5.117 5.117 0 0 1-5.053 5.156v.025z"
          fill="url(#prefix__f)"
        />
        <path
          d="M115.724 32.131h61.339v14.526h-61.339z"
          fill="#8CC14E"
        />
        <path
          d="M146.393 17.95a8.573 8.573 0 0 0-7.92 5.298 8.59 8.59 0 0 0 1.858 9.353 8.566 8.566 0 0 0 9.343 1.86 8.583 8.583 0 0 0 5.292-7.929c0-4.74-3.838-8.583-8.573-8.583zm0 13.536a4.95 4.95 0 0 1-4.941-4.954 4.95 4.95 0 0 1 4.941-4.954 4.95 4.95 0 0 1 4.954 4.942 4.957 4.957 0 0 1-1.452 3.503 4.945 4.945 0 0 1-3.502 1.447v.016z"
          fill="#8CC14E"
        />
        <g
          fill="#8CC14E"
          opacity={0.5}
        >
          <path
            d="M4.687 226.494H5.84v6.555H4.687z"
          />
          <path
            d="M8.54 229.195v1.156H1.991v-1.156z"
          />
        </g>
        <g
          fill="#8CC14E"
          opacity={0.5}
        >
          <path
            d="M138.582 0h1.154v6.555h-1.154z"
          />
          <path
            d="M142.434 2.698v1.156h-6.547V2.698z"
          />
        </g>
        <path
          d="M236.677 237.406a1.417 1.417 0 0 1-.788-1.711.683.683 0 0 0 .03-.157.71.71 0 0 0-1.274-.47.682.682 0 0 0-.08.138c-.286.66-1.021 1-1.709.79a.68.68 0 0 0-.157-.03.709.709 0 0 0-.47 1.274c.044.032.09.06.139.08.659.288.998 1.024.789 1.712a.683.683 0 0 0-.03.157.71.71 0 0 0 1.273.47.682.682 0 0 0 .08-.138c.287-.66 1.022-1 1.71-.79a.68.68 0 0 0 .156.03.709.709 0 0 0 .47-1.275.681.681 0 0 0-.139-.08zM41.95 174.392a1.417 1.417 0 0 1-.789-1.712.683.683 0 0 0 .03-.157.71.71 0 0 0-1.273-.47.682.682 0 0 0-.08.139c-.287.66-1.022.999-1.71.79a.68.68 0 0 0-.156-.031.709.709 0 0 0-.47 1.275c.043.032.09.059.138.08.66.287.999 1.023.79 1.711a.683.683 0 0 0-.031.157.71.71 0 0 0 1.273.47.682.682 0 0 0 .08-.138c.287-.66 1.022-1 1.71-.79a.68.68 0 0 0 .156.031.709.709 0 0 0 .47-1.275.681.681 0 0 0-.138-.08zM204.141 20.952a1.417 1.417 0 0 1-.788-1.712.683.683 0 0 0 .03-.157.71.71 0 0 0-1.273-.47.682.682 0 0 0-.08.138c-.287.66-1.022 1-1.71.79a.68.68 0 0 0-.156-.03.709.709 0 0 0-.47 1.275c.043.032.089.058.138.08.659.287.998 1.023.789 1.711a.683.683 0 0 0-.03.157.71.71 0 0 0 1.273.47.682.682 0 0 0 .08-.138c.287-.66 1.022-1 1.71-.79a.68.68 0 0 0 .156.03.709.709 0 0 0 .47-1.275.681.681 0 0 0-.139-.08z"
          fill="#8CC14E"
          opacity={0.5}
        />
        <ellipse
          cx={249.434}
          cy={96.636}
          fill="#8CC14E"
          opacity={0.5}
          rx={2.311}
          ry={2.314}
        />
        <ellipse
          cx={70.821}
          cy={229.444}
          fill="#8CC14E"
          opacity={0.5}
          rx={2.311}
          ry={2.314}
        />
        <ellipse
          cx={225.698}
          cy={146.652}
          fill="#8CC14E"
          opacity={0.5}
          rx={2.311}
          ry={2.314}
        />
        <ellipse
          cx={26.747}
          cy={29.608}
          fill="#8CC14E"
          opacity={0.5}
          rx={2.311}
          ry={2.314}
        />
        <ellipse
          cx={2.311}
          cy={92.73}
          fill="#8CC14E"
          opacity={0.5}
          rx={2.311}
          ry={2.314}
        />
      </g>
    </svg>
  </div>
  <h1
    className="c2"
  />
  <h2
    className="c3"
  />
</div>
`;
